backend/src/test/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepositoryTest.java

```diff
@@ -666,59 +793,122 @@ void updateUserPrivileges() {
     var groupOrgPrefix = "SR-UNITTEST-TENANT:" + org.getExternalId();
     var groupOrgDefaultName = groupOrgPrefix + ":NO_ACCESS";
     var orgRole = OrganizationRole.ADMIN;
-    var mockUserList = mock(UserList.class);
-    var mockQUserList = mock(UserList.class);
-
     var mockUser = mock(User.class);
-    var mockGroupList = mock(GroupList.class);
     var mockGroup = mock(Group.class);
     var mockGroupProfile = mock(GroupProfile.class);
-    var mockFullGroupList = mock(GroupList.class);
+    var mockGroupProfileToRemove = mock(GroupProfile.class);
     var mockAdminGroup = mock(Group.class);
+    var mockGroupToRemove = mock(Group.class);
     var mockAdminGroupProfile = mock(GroupProfile.class);
+    when(userApi.listUsers(
+            isNull(),
+            isNull(),
+            isNull(),
+            isNull(),
+            eq("profile.login eq \"" + userName + "\""),
+            isNull(),
+            isNull()))
+        .thenReturn(List.of(mockUser));
+    when(mockUser.getId()).thenReturn("1234");
+    when(mockAdminGroup.getId()).thenReturn("adminGID");
+
+    when(userApi.listUserGroups("1234"))
+        .thenReturn(List.of(mockGroup, mockGroupToRemove), List.of(mockGroup, mockAdminGroup));
+    when(mockGroup.getType()).thenReturn(GroupType.OKTA_GROUP);
+    when(mockGroup.getProfile()).thenReturn(mockGroupProfile);
+    when(mockGroupProfile.getName()).thenReturn(groupOrgDefaultName);
+    when(mockGroupToRemove.getType()).thenReturn(GroupType.OKTA_GROUP);
+    when(mockGroupToRemove.getId()).thenReturn("removeGID");
+    when(mockGroupToRemove.getProfile()).thenReturn(mockGroupProfileToRemove);
+    when(mockGroupProfileToRemove.getName()).thenReturn(groupOrgPrefix + "-TO-REMOVE");
+    when(groupApi.listGroups(
+            isNull(),
+            isNull(),
+            isNull(),
+            isNull(),
+            isNull(),
+            eq("profile.name sw \"" + groupOrgPrefix + "\""),
+            isNull(),
+            isNull()))
+        .thenReturn(List.of(mockAdminGroup, mockGroupToRemove));
+    when(mockAdminGroup.getType()).thenReturn(GroupType.OKTA_GROUP);
+    when(mockAdminGroup.getProfile()).thenReturn(mockAdminGroupProfile);
+    when(mockAdminGroupProfile.getName()).thenReturn(groupOrgPrefix + ":" + orgRole);
 
-    when(_client.listUsers(
-            isNull(), isNull(), eq("profile.login eq \"" + userName + "\""), isNull(), isNull()))
+    var actual = _repo.updateUserPrivileges(userName, org, Set.of(), Set.of(orgRole)).orElseThrow();
+    assertThat(actual.getGrantedRoles())
+        .contains(OrganizationRole.ADMIN, OrganizationRole.NO_ACCESS);
+    verify(groupApi).assignUserToGroup("adminGID", "1234");
+    verify(groupApi).unassignUserFromGroup("removeGID", "1234");
+  }
+
+  @Test
+  void updateUserPrivileges_qResultsOnly() {
+    var userName = "fraud@example.com";
```

Comment:
is this to test that using the q param in listUsers works or for something else? I can't tell where we're using the q param here